### PR TITLE
Git Basics: Added a beginner-friendly git handbook to additional resources

### PR DESCRIPTION
**Pull request by theSamyak**
Added beginner-friendly Git fundamentals handbook to additional resources.

## Because
This PR introduces a beginner-friendly Git fundamentals handbook into The Odin Project's additional resources. The handbook is easy for beginners to understand, with clear explanations and step-by-step instructions on Git basics. Its inclusion aims to significantly benefit students by providing them with a comprehensive and accessible resource to learn Git fundamentals effectively. Furthermore, the handbook has gained recognition within the community, having been featured in multiple newsletters and promoted by an official Microsoft page on Twitter, further validating its quality and usefulness.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Thank you for taking the time to make this PR, but we are very intentional about when and where we introduce different Git concepts so this wouldn't really fit with our approach to the subject.

Honestly, we should probably look at removing all additional resources there for that same reason. I would wager most of them predate our shift in how we cover Git.

Is that something you would be interested in doing?

**Comment by theSamyak** [comment]
> Thank you for taking the time to make this PR, but we are very intentional about when and where we introduce different Git concepts so this wouldn't really fit with our approach to the subject.
> 
> Honestly, we should probably look at removing all additional resources there for that same reason. I would wager most of them predate our shift in how we cover Git.
> 
> Is that something you would be interested in doing?



Thank you very much for taking the time to review my pull request and for providing feedback. I truly appreciate the effort you and the team put into maintaining the Odin Project.

I completely understand and respect your decision regarding the inclusion of additional resources in the Git section. Consistency in teaching approach is indeed crucial, and I appreciate your commitment to maintaining it.

If there are any further adjustments you'd like to make, such as removing additional resources or any other contributions you'd like assistance with, please don't hesitate to let me know. I'm more than willing to collaborate further in any capacity that aligns with the project's goals and in whatever capacity I can.

File: markdownlint/docs/TOP004.md [hunk 1]
```diff
@@ -18,7 +18,6 @@ Currently, the required headings depend on the content's type:
 ### Assignment
 #### * (Wildcard: Any subheading at the h4 level)
 ### Knowledge Check
-### Additional Resources
```
Comment (wise-king-sullyman) [inline markdownlint/docs/TOP004.md:21]: We want to keep the additional resources section as required.

File: git/foundations_git/git_basics.md [hunk 2]
```diff
@@ -199,11 +199,3 @@ The following questions are an opportunity to reflect on key topics in this less
 - [How do you commit the files in the staging area and add a descriptive message?](#git-commit)
 - [How do you push your changes to your repository on GitHub?](#git-push)
 - [How do you look at the history of your previous commits?](#git-log)
-
-### Additional resources
-
-This section contains helpful links to related content. It isn't required, so consider it supplemental.
```
Comment (wise-king-sullyman) [inline git/foundations_git/git_basics.md:205]: We'll want to keep this section with the bullet below added as we outline in our [lesson layout guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#lesson-layout)

```
- It looks like this lesson doesn't have any additional resources yet. Help us expand this section by contributing to our curriculum.
```

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Thank you for making these changes! This isn't _quite_ the approach we would want to take though.

**Comment by theSamyak** [comment]
The changes have been made as suggested. Is there anything else I can assist with or any further adjustments required? Please let me know. Thank you.

**Review by xandora — APPROVED** [review]
Just a small issue with a close quote instead of an apostrophe. If there's nothing else to change, I think we can probably merge this one.

**Review by wise-king-sullyman — APPROVED** [review]
Looks perfect :fire: 

Thanks!

**Pull request merged**